packages/matrix-client-server/src/account/password/msisdn/requestToken.ts

```diff
@@ -0,0 +1,162 @@
+  id_server?: string
+  id_access_token?: string
+}
+
+const schema = {
+  client_secret: true,
+  country: true,
+  phone_number: true,
+  next_link: false,
+  send_attempt: true,
+  id_server: false,
+  id_access_token: false
+}
+
+const clientSecretRegex = /^[0-9a-zA-Z.=_-]{6,255}$/
+const validCountryRegex = /^[A-Z]{2}$/ // ISO 3166-1 alpha-2 as per the spec : https://spec.matrix.org/v1.11/client-server-api/#post_matrixclientv3registermsisdnrequesttoken
+const validPhoneNumberRegex = /^\+?[1-9]\d{1,14}$/
+
+const RequestToken = (clientServer: MatrixClientServer): expressAppHandler => {
+  const transport = new SmsSender(clientServer.conf)
+  const verificationTemplate = preConfigureTemplate(
+    fs
+      .readFileSync(`${clientServer.conf.template_dir}/smsVerification.tpl`)
+      .toString(),
+    clientServer.conf,
+    transport
+  )
+  return (req, res) => {
+    jsonContent(req, res, clientServer.logger, (obj) => {
+      validateParameters(res, schema, obj, clientServer.logger, (obj) => {
+        const clientSecret = (obj as RequestTokenArgs).client_secret
+        const sendAttempt = (obj as RequestTokenArgs).send_attempt
+        const country = (obj as RequestTokenArgs).country
+        const phoneNumber = (obj as RequestTokenArgs).phone_number
+        const dst = formatPhoneNumber(phoneNumber, country)
+        const nextLink = (obj as RequestTokenArgs).next_link
+        if (!clientSecretRegex.test(clientSecret)) {
+          send(res, 400, errMsg('invalidParam', 'Invalid client_secret'))
+        } else if (!validCountryRegex.test(country)) {
+          send(res, 400, errMsg('invalidParam', 'Invalid country'))
+          // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+        } else if (nextLink && !isValidUrl(nextLink)) {
+          send(res, 400, errMsg('invalidParam', 'Invalid next_link'))
+        } else if (!validPhoneNumberRegex.test(dst)) {
+          send(res, 400, errMsg('invalidParam', 'Invalid phone number'))
+        } else {
+          clientServer.matrixDb
+            .get('user_threepids', ['user_id'], { address: dst })
+            .then((rows) => {
+              if (rows.length === 0) {
+                send(res, 400, errMsg('threepidNotFound'))
+              } else {
+                clientServer.matrixDb
+                  .get(
+                    'threepid_validation_session',
+                    ['last_send_attempt', 'session_id'],
+                    {
+                      client_secret: clientSecret,
+                      address: dst
+                    }
+                  )
+                  .then((rows) => {
+                    if (rows.length > 0) {
+                      if (sendAttempt === rows[0].last_send_attempt) {
+                        send(res, 200, {
+                          sid: rows[0].session_id,
+                          submit_url: getSubmitUrl(clientServer.conf)
+                        })
+                      } else {
```

Comment:
This "else" has no "send"

Reply:
The send is hidden in the fillTable method that I export from other requestToken files since it's the exact same to avoid copying the same code on multiple files. Should I make this clearer (either by commenting or by copying the fillTable code in all files)

Reply:
Just add a comment